qpylib/encdec.py

```diff
@@ -2,147 +2,146 @@
-        return token
-
-    @staticmethod
-    def __generate_random():
-        """ Returns a string containing a random hash that uses letters, digits and special characters """
-        random_hash = ''.join(
-            (
-                random.choice(string.ascii_letters + string.digits + string.punctuation)
-            )
-            for _ in range(16)
-        )
-        return random_hash
-
-    def __derive_key(self):
-        """ Generates derived key using stored config """
-        return KDF.PBKDF2(self.app_uuid + self.config[self.name]['UUID'],
-                          self.config[self.name]['salt'].encode('utf-8'),
-                          dkLen=32, #32 bytes = 256 bits, max AES key length
-                          count=self.config[self.name]['iterations'])
-
-    def __encrypt_string(self, clear_text_string):
-        """ Returns a string containing an encrypted copy of clear_text_string """
-        aes = AES.new(
-            self.__derive_key(),
-            AES.MODE_CFB,
-            self.config[self.name]['ivz'].encode('utf-8'),
-            segment_size=128)
-        clear_text_padded_string = pad(clear_text_string.encode('utf-8'), AES.block_size)
-        encrypted_bytes = aes.encrypt(clear_text_padded_string)
-        encrypted_hex_bytes = b2a_hex(encrypted_bytes).rstrip()
-        return encrypted_hex_bytes.decode('utf-8')
-
-    def __decrypt_string(self, encrypted_string):
-        """ Returns a string containing a decrypted copy of encrypted_string.
-            encrypted_string must be the result of a previous call to encrypt(). """
-        aes = AES.new(
-            self.__derive_key(),
-            AES.MODE_CFB,
-            self.config[self.name]['ivz'].encode('utf-8'),
-            segment_size=128)
-        encrypted_hex_bytes = encrypted_string.encode('utf-8')
-        encrypted_bytes = a2b_hex(encrypted_hex_bytes)
-        decrypted_bytes = unpad(aes.decrypt(encrypted_bytes), AES.block_size)
-        return decrypted_bytes.decode('utf-8')
+        if os.path.isfile(self.config_store_path):
+            self.config = self._read_config()
+        else:
+            self.config = {}
 
     def encrypt(self, clear_text):
-        """ Encrypts a clear text secret """
+        ''' Encrypts clear_text using the latest engine version.
+            Stores the encrypted value using the data provided to the init function,
+            i.e. in a store file named 'user'_e.db under label 'name'.
+            Returns the encrypted value.
+        '''
+        self._reset_config_if_required()
+
         try:
-            self.config[self.name]['version'] = Encryption.engine_version
-            self.config[self.name]['secret'] = self.__encrypt_string(clear_text)
-            self.__save_config()
-            return self.config[self.name]['secret']
+            self.config[self.name]['secret'] = self._get_latest_engine().encrypt(clear_text)
+        except Exception as error:
```

Comment:
The various crypto packages are rubbish at publishing what exceptions they raise. Rather than having a long and possibly incomplete list, I took the simpler approach. We don't actually care what the exception type is, we just catch it and raise our own exception, and it seems neater.